Fix the structure of sublists in `changes.md`

CyrilleB79 opened the pull request:
### Link to issue number:

Fixes #16662

### Summary of the issue:
When the documentation was written in .t2t format, the format for sub-lists was more tolerant regarding the size of the indentation of the sub-list. Most sub-lists were indented 2 spaces, but some of them were only indented 1 space. Now that we have transitioned to .md format, this atypical indentation causes them not to be recognized as a sub-list when converted to HTML.

### Description of user facing changes
* The sub-lists in the HTML change log should now appear as before the t2t to md transition.
* Translators should not be impacted by this change. I.e. modified content belong to an already translated change log and translators should not have to re-translate old parts of the change log.

### Description of development approach
* Searched for lines beginning with an even number of space (regexp: "`(?<=\n) (  )*(?=[^ ])`") and manually fixed each found line as well as the subsequent ones, i.e. other sub-list items and sub-sub-list items
* Did not modify lines beginning with space but no list bullet, because it would have an impact on translators work; moreover these ones do not seem to cause HTML formatting issues.

One line beginning with " -  " remains unfixed.
I cannot modify it without impacting translators' work.

### Testing strategy:
* Used the xliff generation script to check that no translator's change will heppen. See https://github.com/nvaccess/nvda/pull/17741#issuecomment-2747459979 for details.

### Known issues with pull request:
* If some translators have a translated change log in SVN that has not yet been migrated to Crowdin, it will probably not be possible anymore to migrate SVN content, i.e. the change log will have to be re-translated from scratch. Though, I guess that all translators that would have needed to pre-translate their translation should already have migrated.
* If #17735 is implemented, this PR may be more difficult to merge; thus, it would be better to merge it before implementing #17735.
### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

<!-- Please keep the following -->
@coderabbitai summary

Commit:
Fix the structure of sublists in `changes.md`

CyrilleB79 committed:
Some more struct fixes

seanbudd commented:
@CyrilleB79 - I think it's unlikely you will be able to change the structure without translating needing to retranslate. You may need to set up your own crowdin instance to test if you want to confirm

Commit:
Merge branch 'master' into fixChangeLogLists

CyrilleB79 commented:
@seanbudd to test that no strings were changed, I have done as follows:

I have looked at the action https://github.com/nvaccess/nvda/blob/master/.github/workflows/regenerate_english_userDocs_translation_source.yml.

From it, I have extracted the powershell script which updates the xliff files in a specific [action.ps1.txt](https://github.com/user-attachments/files/19426747/action.ps1.txt) (txt extension to be removed).
In this file, I have replaced the commit provided by GitHub by `master` so that I make the diff between master and the changes in this PR.
Moreover, I have had to copy `markdownTranslate.py` from `source` to `user_docs`; I do not know exactly why the action was able to find it there.; I have not investigated this..



I have run it:
`powershell -ExecutionPolicy Bypass -File action.ps1`

And looked at the output:


```
user_docs/en/changes.md has changed
Updating user_docs/en/changes.xliff with changes from user_docs/en/changes.md
Generating updated xliff file user_docs/en/changes.xliff.temp from user_docs/en/changes.xliff and user_docs/en/changes.md...
Generating markdown file user_docs\en\generated_iisjgg_h.md from user_docs/en/changes.xliff...
Generated markdown file with 5346 total lines, 4569 translatable strings, and 0 translated strings. Ignoring 0 bad translated strings
Extracting skeleton from user_docs/en/changes.xliff to user_docs\en\extracted_dcfj14r8.skel...
Extracted skeleton to user_docs\en\extracted_dcfj14r8.skel
Creating updated skeleton file user_docs\en\updated_2u_2iiiy.skel from user_docs\en\extracted_dcfj14r8.skel with changes from user_docs\en\generated_iisjgg_h.md to user_docs/en/changes.md...
Updated skeleton file with 0 added lines (0 translation placeholders), 0 removed lines (0 translation placeholders), and 5346 unchanged lines (4569 translation placeholders)
Generating xliff file user_docs/en/changes.xliff.temp from user_docs/en/changes.md and user_docs\en\updated_2u_2iiiy.skel...
Including Github raw URL: https://raw.githubusercontent.com/nvaccess/nvda/8d6487ad545c57d0a8802c3e2d8ac438b9481d3c/user_docs/en/changes.md
Generated xliff file with 4569 translatable strings
Generated updated xliff file user_docs/en/changes.xliff.temp
Renaming user_docs/en/changes.xliff.temp to user_docs/en/changes.xliff
```

With the following information:
`0 added lines (0 translation placeholders), 0 removed lines (0 translation placeholders)`
We can confirm that no translation change will be pushed.

Then, to double check, I have modified one line in `user_docs/en/changes.md`, and re-run `action.ps1`:

The result was as follows:

```
user_docs/en/changes.md has changed
Updating user_docs/en/changes.xliff with changes from user_docs/en/changes.md
Generating updated xliff file user_docs/en/changes.xliff.temp from user_docs/en/changes.xliff and user_docs/en/changes.md...
Generating markdown file user_docs\en\generated_b72n6ipt.md from user_docs/en/changes.xliff...
Generated markdown file with 5346 total lines, 4569 translatable strings, and 0 translated strings. Ignoring 0 bad translated strings
Extracting skeleton from user_docs/en/changes.xliff to user_docs\en\extracted_wk6tdntn.skel...
Extracted skeleton to user_docs\en\extracted_wk6tdntn.skel
Creating updated skeleton file user_docs\en\updated_hvrov0vo.skel from user_docs\en\extracted_wk6tdntn.skel with changes from user_docs\en\generated_b72n6ipt.md to user_docs/en/changes.md...
Updated skeleton file with 1 added lines (1 translation placeholders), 1 removed lines (1 translation placeholders), and 5345 unchanged lines (4568 translation placeholders)
Generating xliff file user_docs/en/changes.xliff.temp from user_docs/en/changes.md and user_docs\en\updated_hvrov0vo.skel...
Including Github raw URL: https://raw.githubusercontent.com/nvaccess/nvda/8d6487ad545c57d0a8802c3e2d8ac438b9481d3c/user_docs/en/changes.md
Generated xliff file with 4569 translatable strings
Generated updated xliff file user_docs/en/changes.xliff.temp
Renaming user_docs/en/changes.xliff.temp to user_docs/en/changes.xliff
```

With the following information:
`1 added lines (1 translation placeholders), 1 removed lines (1 translation placeholders)`
We can confirm that a translation change would have been pushed, confirming that the script that I have modified still captures translation changes.

Maybe @michaelDCurran could confirm that this test is enough?

CyrilleB79 commented:
@seanbudd in addition to my previous tests, if you feel this modification too risky:
1. I can provide a PR changing only one line. When it lands to beta and xliff is generated, we can confirm that the line does not be re-translated and then merge the current PR in beta to fix the remaining lines.
2. Or we can merge this PR as is. And when branching to beta and xliff file is generated, you could check that old items are not to be re-translated. In case of issue, you should immediately revert this PR in beta.

What do you think?

wmhn1872265132 commented:
>In this file, I have replaced the commit provided by GitHub by master so that I make the diff between master and the changes in this PR.
Moreover, I have had to copy markdownTranslate.py from source to user_docs; I do not know exactly why the action was able to find it there.; I have not investigated this..

See https://github.com/nvaccess/nvda/pull/17712#issuecomment-2673059285
